### PR TITLE
Merge `templates/projects/list.tmpl` and `templates/repo/projects/list.tmpl` together

### PR DESCRIPTION
**Pull request by earl-warren**
(cherry picked from commit 473862a1d599382ca022482e2e044025872d240b)

Refs: https://codeberg.org/forgejo/forgejo/pulls/1126

### CODE REVIEW

**Review by wxiaoguang — APPROVED** [review]

File: templates/projects/list.tmpl [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{if .CanWriteProjects}}
+{{if and $.CanWriteProjects (not $.Repository.IsArchived)}}
```
Comment (delvh) [inline templates/projects/list.tmpl:1]: Isn't that a regression for org/user projects?
Do they have a `Repository` attribute?

Comment (delvh) [reply templates/projects/list.tmpl:1]: Okay, I just tested it.
For whatever reason, it works.

**Review by delvh — APPROVED** [review]
Thanks to both of you, earl and louis, now I have less work to do (https://github.com/go-gitea/gitea/issues/25123#issuecomment-1581587465).
I don't know why the solution works but I'm fine with it working. I didn't expect it to be that easy, it looked like that would need a lot more changes…
The only thing that remains for my point
> Merging the separate project templates together

Is the actual display of one project (`templates/projects/view.tmpl` vs `templates/repo/projects/view.tmpl`). Merging Creation/Edits and List are already done then…

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
It also fixed some bugs.

**Comment by lunny** [comment]
The bug could be backport to v1.20 but I don't whether the refacto can do also.

**Comment by delvh** [comment]
Which bug did it fix?

File: templates/projects/list.tmpl (hunk 1 above)
Comment (lunny) [reply templates/projects/list.tmpl:1]: wait. $.Repository maybe nil?

**Review by lunny — DISMISSED** [review]

File: templates/projects/list.tmpl (hunk 1 above)
Comment (delvh) [reply templates/projects/list.tmpl:1]: No, for some reason it isn't.
As I said, I tested it, it works.
I don't know either why.

Comment (lunny) [reply templates/projects/list.tmpl:1]: But it's really wrong? Why not fix it?

Comment (wxiaoguang) [reply templates/projects/list.tmpl:1]: It's not wrong.

https://pkg.go.dev/text/template#Template.Option

Comment (delvh) [reply templates/projects/list.tmpl:1]: That's the weird thing:
I think it's invalid too, but for some weird reason it works.
So, let's leave the clean up of that for another PR?

Comment (lunny) [reply templates/projects/list.tmpl:1]: We need to get to know what happened before merging this.

**Comment by denyskon** [comment]
Why do we backport a refactor? 😳

**Comment by lunny** [comment]
> 

It checks the archive repositories to prevent to display the `new project` button

**Pull request merged**